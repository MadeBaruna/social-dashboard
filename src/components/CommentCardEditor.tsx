import React, { Component, SyntheticEvent } from 'react';
import {
  Card,
  Button,
  Form,
  Input,
  TextArea,
  InputOnChangeData,
  TextAreaProps,
} from 'semantic-ui-react';
import { DataProxy } from 'apollo-cache';
import { MutationFn, FetchResult, Mutation } from 'react-apollo';

import { GetPost as GetPostQuery } from '../graphql/queries/GetPost';
import { CreateComment as CreateCommentMutation } from '../graphql/mutations/CreateComment';
import { UpdateComment as UpdateCommentMutation } from '../graphql/mutations/UpdateComment';
import {
  CreateComment,
  CreateCommentVariables,
} from '../graphql/mutations/__generated__/CreateComment';
import {
  UpdateComment,
  UpdateCommentVariables,
} from '../graphql/mutations/__generated__/UpdateComment';
import {
  GetPost,
  GetPostVariables,
} from '../graphql/queries/__generated__/GetPost';

interface IProps {
  postId: number;
  id: number;
  name: string;
  email: string;
  body: string;
  cancelEditing?: () => void;
  updateCard?: (name: string, email: string, body: string) => void;
}

interface IState {
  name?: string;
  email?: string;
  body?: string;
}

class CommentCardEditor extends Component<IProps, IState> {
  public static defaultProps = {
    id: -1,
    name: '',
    email: '',
    body: '',
  };

  public state = {
    name: this.props.name,
    email: this.props.email,
    body: this.props.body,
  };

  public render() {
    const { id, cancelEditing } = this.props;
    const { name, email, body } = this.state;

    return (
      <Card fluid>
        <Card.Content>
          <Form>
            <Form.Field>
              <Input
                label="Name"
                placeholder="Your name"
                value={name}
                onChange={this.onInputChange('name')}
              />
            </Form.Field>
            <Form.Field>
              <Input
                label="Email"
                placeholder="Your email"
                value={email}
                onChange={this.onInputChange('email')}
              />
            </Form.Field>
            <Form.Field>
              <TextArea
                autoHeight
                placeholder="What do you think?"
                value={body}
                onInput={this.onTextAreaChange}
              />
            </Form.Field>
          </Form>
        </Card.Content>
        <Card.Content extra textAlign="right">
          <Mutation mutation={CreateCommentMutation} update={this.update}>
            {(CreateCommentFunction, { loading: loadingCreate }) => (
              <Mutation mutation={UpdateCommentMutation}>
                {(UpdateCommentFunction, { loading: loadingUpdate }) => (
                  <Button
                    loading={loadingCreate || loadingUpdate}
                    primary
                    onClick={this.submit(
                      CreateCommentFunction,
                      UpdateCommentFunction,
                    )}
                  >
                    {id === -1 ? 'Submit' : 'Save'} Comment
                  </Button>
                )}
              </Mutation>
            )}
          </Mutation>

          {cancelEditing && <Button onClick={cancelEditing}>Cancel</Button>}
        </Card.Content>
      </Card>
    );
  }

  private onInputChange = (field: 'name' | 'email') => (
    _: SyntheticEvent,
    data: InputOnChangeData,
  ) => {
    this.setState({ [field]: data.value });
  }

  private onTextAreaChange = (_: SyntheticEvent, data: TextAreaProps) => {
    this.setState({ body: data.value as string });
  }

  private submit = (
    createMutationFunction: MutationFn<CreateComment, CreateCommentVariables>,
    updateMutationFunction: MutationFn<UpdateComment, UpdateCommentVariables>,
  ) => async () => {
    const { postId, id, updateCard, cancelEditing } = this.props;
    const { name, email, body } = this.state;

    if (id === -1) {
      createMutationFunction({
        variables: {
          postId,
          name,
          email,
          body,
        },
      });
    } else {
      await updateMutationFunction({
        variables: {
          id,
          name,
          email,
          body,
        },
      });

      if (updateCard && cancelEditing) {
        updateCard(name, email, body);
        cancelEditing();
      }
    }
  }

  private update = (
    cache: DataProxy,
    { data }: FetchResult<CreateComment, Record<string, any>>,
  ) => {
    if (!data) {
      return;
    }

    const { postId } = this.props;
    const postDetail = cache.readQuery<GetPost, GetPostVariables>({
      query: GetPostQuery,
      variables: { id: postId },
    });

    if (!postDetail) {
      return;
    }

    const { comment } = data;
    const { post } = postDetail;

    const updatedComments = [...post.comments, comment];
    post.comments = updatedComments;

    cache.writeQuery({
      query: GetPostQuery,
      variables: {
        id: postId,
      },
      data: { post },
    });

    this.setState({ name: '', email: '', body: '' });
  }
}

export default CommentCardEditor;
