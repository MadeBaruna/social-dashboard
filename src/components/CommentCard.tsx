import React, { Component } from 'react';
import { DataProxy } from 'apollo-cache';
import { MutationFn, Mutation, FetchResult } from 'react-apollo';
import { Card, Button } from 'semantic-ui-react';

import CommentCardEditor from './CommentCardEditor';
import { DeleteComment as DeleteCommentMutation } from '../graphql/mutations/DeleteComment';
import {
  DeleteComment,
  DeleteCommentVariables,
} from '../graphql/mutations/__generated__/DeleteComment';
import { GetPost as GetPostQuery } from '../graphql/queries/GetPost';
import {
  GetPost,
  GetPostVariables,
} from '../graphql/queries/__generated__/GetPost';

interface IProps {
  postId: number;
  id: number;
  name: string;
  email: string;
  body: string;
}

interface IState {
  isEditing: boolean;
  name: string;
  email: string;
  body: string;
}

class CommentCard extends Component<IProps, IState> {
  public state = {
    isEditing: false,
    name: this.props.name,
    email: this.props.email,
    body: this.props.body,
  };

  public render() {
    const { name, email, body, isEditing } = this.state;

    if (isEditing) {
      return (
        <CommentCardEditor
          {...this.props}
          updateCard={this.updateCard}
          cancelEditing={this.setEditing(false)}
        />
      );
    }

    return (
      <Card fluid>
        <Card.Content>
          <Card.Header>{name}</Card.Header>
          <Card.Meta>{email}</Card.Meta>
          {body.split('\n').map((text, key) => (
            <span key={key}>
              {text}
              <br />
            </span>
          ))}
        </Card.Content>
        <Card.Content extra textAlign="right">
          <Button icon="pencil" onClick={this.setEditing(true)} />
          <Mutation mutation={DeleteCommentMutation} update={this.update}>
            {(DeleteCommentFunction, { loading }) => (
              <Button
                icon="trash"
                loading={loading}
                onClick={this.delete(DeleteCommentFunction)}
              />
            )}
          </Mutation>
        </Card.Content>
      </Card>
    );
  }

  private setEditing = (isEditing: boolean) => () => {
    this.setState({ isEditing });
  }

  private updateCard = (name: string, email: string, body: string) => {
    this.setState({ name, email, body });
  }

  private delete = (
    mutation: MutationFn<DeleteComment, DeleteCommentVariables>,
  ) => async () => {
    const { id } = this.props;

    await mutation({
      variables: {
        id,
      },
    });
  }

  private update = (
    cache: DataProxy,
    { data }: FetchResult<DeleteComment, Record<string, any>>,
  ) => {
    if (!data) {
      return;
    }

    const { postId, id } = this.props;

    let postDetail;
    try {
      postDetail = cache.readQuery<GetPost, GetPostVariables>({
        query: GetPostQuery,
        variables: { id: postId },
      });
    } catch (err) {
      return;
    }

    if (!postDetail) {
      return;
    }

    const deletedCommentId = data.deleteComment.id;
    const { post } = postDetail;

    const updatedComments = post.comments.filter(
      (comment) => comment.id !== deletedCommentId,
    );

    post.comments = updatedComments;

    cache.writeQuery({
      query: GetPostQuery,
      variables: {
        id: postId,
      },
      data: { post },
    });
  }
}

export default CommentCard;
