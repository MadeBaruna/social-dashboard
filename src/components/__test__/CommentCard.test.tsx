import React from 'react';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import { Input, TextArea, Button } from 'semantic-ui-react';

import CommentCard from '../CommentCard';
import CommentCardEditor from '../CommentCardEditor';
import PostCardEditor from '../PostCardEditor';

it('renders CommentCard components', () => {
  const wrapper = shallow(
    <CommentCard
      postId={1}
      id={1}
      name="name"
      email="email"
      body={
        'qui consequuntur ducimus\npossimus quisquam amet similique\nsuscipit porro ipsam amet'
      }
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders CommentCardEditor if editing', () => {
  const wrapper = shallow(
    <CommentCard
      postId={1}
      id={1}
      name="name"
      email="email"
      body={
        'qui consequuntur ducimus\npossimus quisquam amet similique\nsuscipit porro ipsam amet'
      }
    />,
  );

  wrapper
    .find(Button)
    .at(0)
    .simulate('click');
  expect(wrapper.find(CommentCardEditor)).toHaveLength(1);
});

it('Set input value when editing', () => {
  const wrapper = shallow(
    <CommentCard postId={1} id={1} name="name" email="email" body="body" />,
  );

  wrapper
    .find(Button)
    .at(0)
    .simulate('click');

  const inputs = wrapper.dive().find(Input);
  expect(inputs.at(0).props().value).toEqual('name');
  expect(inputs.at(1).props().value).toEqual('email');
  expect(
    wrapper
      .dive()
      .find(TextArea)
      .props().value,
  ).toEqual('body');
});

it('Keep value if edit is cancelled', () => {
  const wrapper = shallow(
    <CommentCard postId={1} id={1} name="name" email="email" body="body" />,
  );

  wrapper
    .find(Button)
    .at(0)
    .simulate('click');

  wrapper.update();

  wrapper
    .dive()
    .find(Button)
    .at(0)
    .simulate('click');

  expect(wrapper.state()).toEqual({
    name: 'name',
    email: 'email',
    body: 'body',
    isEditing: false,
  });
});
