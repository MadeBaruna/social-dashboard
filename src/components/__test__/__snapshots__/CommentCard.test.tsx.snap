// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentCard components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentCard
    body="qui consequuntur ducimus
possimus quisquam amet similique
suscipit porro ipsam amet"
    email="email"
    id={1}
    name="name"
    postId={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CardContent>
          <CardHeader>
            name
          </CardHeader>
          <CardMeta>
            email
          </CardMeta>
          <span>
            qui consequuntur ducimus
            <br />
          </span>
          <span>
            possimus quisquam amet similique
            <br />
          </span>
          <span>
            suscipit porro ipsam amet
            <br />
          </span>
        </CardContent>,
        <CardContent
          extra={true}
          textAlign="right"
        >
          <Button
            as="button"
            icon="pencil"
            onClick={[Function]}
          />
          <Mutation
            mutation={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "DeleteComment",
                    },
                    "operation": "mutation",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                              },
                            },
                          ],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "deleteComment",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [
                      Object {
                        "defaultValue": undefined,
                        "directives": Array [],
                        "kind": "VariableDefinition",
                        "type": Object {
                          "kind": "NonNullType",
                          "type": Object {
                            "kind": "NamedType",
                            "name": Object {
                              "kind": "Name",
                              "value": "Int",
                            },
                          },
                        },
                        "variable": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                        },
                      },
                    ],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 87,
                  "start": 0,
                },
              }
            }
            update={[Function]}
          >
            [Function]
          </Mutation>
        </CardContent>,
      ],
      "fluid": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader>
              name
            </CardHeader>,
            <CardMeta>
              email
            </CardMeta>,
            Array [
              <span>
                qui consequuntur ducimus
                <br />
              </span>,
              <span>
                possimus quisquam amet similique
                <br />
              </span>,
              <span>
                suscipit porro ipsam amet
                <br />
              </span>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "name",
            },
            "ref": null,
            "rendered": "name",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "email",
            },
            "ref": null,
            "rendered": "email",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "qui consequuntur ducimus",
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "qui consequuntur ducimus",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "possimus quisquam amet similique",
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "possimus quisquam amet similique",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "suscipit porro ipsam amet",
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "suscipit porro ipsam amet",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Button
              as="button"
              icon="pencil"
              onClick={[Function]}
            />,
            <Mutation
              mutation={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "DeleteComment",
                      },
                      "operation": "mutation",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "deleteComment",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [
                        Object {
                          "defaultValue": undefined,
                          "directives": Array [],
                          "kind": "VariableDefinition",
                          "type": Object {
                            "kind": "NonNullType",
                            "type": Object {
                              "kind": "NamedType",
                              "name": Object {
                                "kind": "Name",
                                "value": "Int",
                              },
                            },
                          },
                          "variable": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                          },
                        },
                      ],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 87,
                    "start": 0,
                  },
                }
              }
              update={[Function]}
            >
              [Function]
            </Mutation>,
          ],
          "extra": true,
          "textAlign": "right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "icon": "pencil",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": [Function],
              "mutation": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "DeleteComment",
                    },
                    "operation": "mutation",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                              },
                            },
                          ],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "deleteComment",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [
                      Object {
                        "defaultValue": undefined,
                        "directives": Array [],
                        "kind": "VariableDefinition",
                        "type": Object {
                          "kind": "NonNullType",
                          "type": Object {
                            "kind": "NamedType",
                            "name": Object {
                              "kind": "Name",
                              "value": "Int",
                            },
                          },
                        },
                        "variable": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                        },
                      },
                    ],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 87,
                  "start": 0,
                },
              },
              "update": [Function],
            },
            "ref": null,
            "rendered": [Function],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CardContent>
            <CardHeader>
              name
            </CardHeader>
            <CardMeta>
              email
            </CardMeta>
            <span>
              qui consequuntur ducimus
              <br />
            </span>
            <span>
              possimus quisquam amet similique
              <br />
            </span>
            <span>
              suscipit porro ipsam amet
              <br />
            </span>
          </CardContent>,
          <CardContent
            extra={true}
            textAlign="right"
          >
            <Button
              as="button"
              icon="pencil"
              onClick={[Function]}
            />
            <Mutation
              mutation={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "DeleteComment",
                      },
                      "operation": "mutation",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "deleteComment",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [
                        Object {
                          "defaultValue": undefined,
                          "directives": Array [],
                          "kind": "VariableDefinition",
                          "type": Object {
                            "kind": "NonNullType",
                            "type": Object {
                              "kind": "NamedType",
                              "name": Object {
                                "kind": "Name",
                                "value": "Int",
                              },
                            },
                          },
                          "variable": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                          },
                        },
                      ],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 87,
                    "start": 0,
                  },
                }
              }
              update={[Function]}
            >
              [Function]
            </Mutation>
          </CardContent>,
        ],
        "fluid": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardHeader>
                name
              </CardHeader>,
              <CardMeta>
                email
              </CardMeta>,
              Array [
                <span>
                  qui consequuntur ducimus
                  <br />
                </span>,
                <span>
                  possimus quisquam amet similique
                  <br />
                </span>,
                <span>
                  suscipit porro ipsam amet
                  <br />
                </span>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "name",
              },
              "ref": null,
              "rendered": "name",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "email",
              },
              "ref": null,
              "rendered": "email",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "qui consequuntur ducimus",
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "qui consequuntur ducimus",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "possimus quisquam amet similique",
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "possimus quisquam amet similique",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "suscipit porro ipsam amet",
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "suscipit porro ipsam amet",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Button
                as="button"
                icon="pencil"
                onClick={[Function]}
              />,
              <Mutation
                mutation={
                  Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": Object {
                          "kind": "Name",
                          "value": "DeleteComment",
                        },
                        "operation": "mutation",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "value": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                  },
                                },
                              ],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "deleteComment",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [
                          Object {
                            "defaultValue": undefined,
                            "directives": Array [],
                            "kind": "VariableDefinition",
                            "type": Object {
                              "kind": "NonNullType",
                              "type": Object {
                                "kind": "NamedType",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "Int",
                                },
                              },
                            },
                            "variable": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                            },
                          },
                        ],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 87,
                      "start": 0,
                    },
                  }
                }
                update={[Function]}
              >
                [Function]
              </Mutation>,
            ],
            "extra": true,
            "textAlign": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "icon": "pencil",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": [Function],
                "mutation": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "DeleteComment",
                      },
                      "operation": "mutation",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "deleteComment",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [
                        Object {
                          "defaultValue": undefined,
                          "directives": Array [],
                          "kind": "VariableDefinition",
                          "type": Object {
                            "kind": "NonNullType",
                            "type": Object {
                              "kind": "NamedType",
                              "name": Object {
                                "kind": "Name",
                                "value": "Int",
                              },
                            },
                          },
                          "variable": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                          },
                        },
                      ],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 87,
                    "start": 0,
                  },
                },
                "update": [Function],
              },
              "ref": null,
              "rendered": [Function],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
